'openssl' include in 'oiio/src/libOpenImageIO/imagebufalgo.cpp' file generates deprecation compilation errors on Mac Os X 10.7.5.

/Users/kelsolaar/Documents/Development/ThirdParty/oiio/src/libOpenImageIO/imagebufalgo.cpp:1569:5: error: 'SHA1_Init' is deprecated [-Werror,-Wdeprecated-declarations]
    SHA1_Init (&sha);
    ^
/Users/kelsolaar/Documents/Development/ThirdParty/oiio/src/libOpenImageIO/imagebufalgo.cpp:1575:17: error: 'SHA1_Update' is deprecated [-Werror,-Wdeprecated-declarations]
                SHA1_Update (&sha, src.pixeladdr (roi.xbegin, y, z),
                ^
/Users/kelsolaar/Documents/Development/ThirdParty/oiio/src/libOpenImageIO/imagebufalgo.cpp:1580:17: error: 'SHA1_Update' is deprecated [-Werror,-Wdeprecated-declarations]
                SHA1_Update (&sha, &tmp[0], (unsigned int) scanline_bytes_(y1-y));
                ^
/Users/kelsolaar/Documents/Development/ThirdParty/oiio/src/libOpenImageIO/imagebufalgo.cpp:1587:9: error: 'SHA1_Update' is deprecated [-Werror,-Wdeprecated-declarations]
        SHA1_Update (&sha, extrainfo.c_str(), extrainfo.size());
        ^
/Users/kelsolaar/Documents/Development/ThirdParty/oiio/src/libOpenImageIO/imagebufalgo.cpp:1591:5: error: 'SHA1_Final' is deprecated [-Werror,-Wdeprecated-declarations]
    SHA1_Final (md, &sha);
    ^
/Users/kelsolaar/Documents/Development/ThirdParty/oiio/src/libOpenImageIO/imagebufalgo.cpp:1691:5: error: 'SHA1_Init' is deprecated [-Werror,-Wdeprecated-declarations]
    SHA1_Init (&sha);
    ^
/Users/kelsolaar/Documents/Development/ThirdParty/oiio/src/libOpenImageIO/imagebufalgo.cpp:1693:9: error: 'SHA1_Update' is deprecated [-Werror,-Wdeprecated-declarations]
        SHA1_Update (&sha, results[b].c_str(), results[b].size());
        ^
/Users/kelsolaar/Documents/Development/ThirdParty/oiio/src/libOpenImageIO/imagebufalgo.cpp:1695:9: error: 'SHA1_Update' is deprecated [-Werror,-Wdeprecated-declarations]
        SHA1_Update (&sha, extrainfo.c_str(), extrainfo.size());
        ^
/Users/kelsolaar/Documents/Development/ThirdParty/oiio/src/libOpenImageIO/imagebufalgo.cpp:1698:5: error: 'SHA1_Final' is deprecated [-Werror,-Wdeprecated-declarations]
    SHA1_Final (md, &sha);
    ^
9 errors generated.
make[3]: *_\* [libOpenImageIO/CMakeFiles/OpenImageIO.dir/imagebufalgo.cpp.o] Error 1
make[2]: **\* [libOpenImageIO/CMakeFiles/OpenImageIO.dir/all] Error 2
make[1]: **\* [all] Error 2
make: **\* [cmake] Error 2

--- a/src/libOpenImageIO/imagebufalgo_compare.cpp
+++ b/src/libOpenImageIO/imagebufalgo_compare.cpp
@@ -53,6 +53,11 @@
 #include "thread.h"
 
 #ifdef USE_OPENSSL
+#ifdef __APPLE__
+// Newer OSX releaes mark OpenSSL functions as deprecated, in favor of
+// CDSA.  Make the warnings stop.
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations" 
+#endif
 #include <openssl/sha.h>
 #endif
 

